test/e2e/tls_test.go

```diff
@@ -65,14 +68,14 @@ func TestAntreaApiserverTLSConfig(t *testing.T) {
 		apiserver     int
 		apiserverStr  string
 	}{
-		{"ControllerApiserver", controllerPodName, controllerContainerName, apis.AntreaControllerAPIPort, "Controller"},
+		{"ControllerApiserver", agentPodName, agentContainerName, apis.AntreaControllerAPIPort, "Controller"},
```

Comment:
I do think the test would make more sense if the client could be a separate toolbox Pod. openssl is not available in the toolbox Pod, but I wonder if we could just use curl with the appropriate flags (`--tlsv1.2`, `--tls-max`, etc.)?

Reply:
since we already have openssl in antrea-agent pod so wouldn't creating new toolbox pod be an overhead, because in antrea-agent image we are using the ovs image and ovs has openssl installed.

Reply:
@antoninbas 
Both ways, openssl and curl can be used test Cipher Suite and TLSVersion config on Antrea apiserver, Controller side or Agent side,
For checking basic TLS version and cipher suites , curl might be more convenient alongside toolbox pod however ovs needs ssl, hence can not be removed and size benefit of not having openssl is not there !

Reply:
sure would make the necessary changes.

Reply:
@antoninbas i tried this but it didn't work, when using curl to connect to the API server we have to additionally provide the account token and necessary permissions.

Reply:
`curl -k -v --head --tls-max 1.2 --tlsv1.2 https://172.18.0.3:10350/` works fine for me.
While we do get an HTTP 403 code (forbidden), it is irrelevant for the test: the curl command is still considered successful.
By using the verbose flag (`-v`), the output contains the information we are looking for:
```
SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
```